container/openejb-core/src/main/java/org/apache/openejb/assembler/classic/Assembler.java

```diff
@@ -2259,22 +2259,6 @@ public void destroyApplication(final String filePath) throws UndeployException,
         }
     }
 
-    public void destroyApplication(final AppContext appContext) throws UndeployException {
```

Comment:
I'd never remove a public method like this, as it's impossible to tell if some other tool is using this.

Reply:
I'd actually just write the 'missing' test

Reply:
This method will never work. It will always fail with IllegalStateException. Deployed applications are kept in a map and the key is the path of the deployed application. Path information is not available in AppContext and in this method application is searched by appContext.getId() which is not app path. 

That is the main reason I suggest this method to be removed.

Reply:
I agree it's useless. But I'd still just deprecate it for the next release, then remove it after. I just can't guarantee that some 3rd party has not written code around it, even if it throws an error.